packages/common/src/config/logger.ts

```diff
@@ -1,6 +1,20 @@
-import { Logger, LoggerOptions, createLogger, format, transports } from 'winston';
+import { LoggerOptions, createLogger, format, transports } from 'winston';
 import { LoggerConfig } from './config.js';
 
+export type loggerFunction = (...messages: any[]) => void;
+
+export type Logger = {
+	debug: loggerFunction;
+	warn: loggerFunction;
+	info: loggerFunction;
+	error: loggerFunction;
+};
```

Comment:
Does a default winston logger match this type?

I like this abstraction I just want to make sure winston (which is not the only node logger but is common) can be used as a logger without our own customizations. Don't need further changes for this PR, its just a place we could make further improvements to be more compliant with the JS community.

Reply:
This is a custom `Logger` type different from winston's type. The purpose of this is to provide 4 standard loggerFunctions, each of these implements a winston function. So, in case we require to use a different type of node logger we will need to implement these 4 logger functions.